### DISCUSSION TITLE
Anyone have experience with digital download products

### DISCUSSION THREAD

**Discussion by c00lcoder** [saleor/saleor#12238] [General]
I'm curious if anyone has used saleor to offer digital download products. For example, a storefront where customers can pay a-la-cart for digital products or access X amount of products via a subscription.

**Comment by timuric** [comment]
@c00lcoder You can store links in private metadata and serve that to customers via email link or a page

**Reply by c00lcoder** [reply]
Okay, that could work. Is it possible to integrate a function that dynamically creates the links for the image and stores it in the private metadata? The goal would be to have a unique link (masked/pre-signed) per transaction, so there isn't a discernable url pattern. For example, images being served from aws s3 or dropbox

**Comment by rrrnld** [comment]
We are currently brainstorming this; the way we're planning to solve this is through an app that generates a unique token when an order is paid for, and stores it in the private order metadata. It then publically serves requests like `http://$public_url_to_app/download/$token`, queries for confirmed orders with this token, and acts as a proxy to stream files from private urls that are attached to any product in the order.

**Reply by c00lcoder** [reply]
Ooh, that sounds promising! I'm still figuring out saleor but if you all want a tester for this functionality, I'd be happy to help
